packages/@sanity/desk-tool/src/panes/documentPane/timeline/timeline.tsx

```diff
@@ -27,113 +26,91 @@ interface TimelineProps {
-              menu.
-            </Text>
-          </Stack>
-        )}
-        <ol className={styles.list} ref={listRef}>
-          {timeline.mapChunks((chunk) => {
-            const isSelectionTop = topSelection === chunk
-            const isSelectionBottom = bottomSelection === chunk
-
-            if (isSelectionTop) {
-              state = 'withinSelection'
-            }
-
-            if (isSelectionBottom) {
-              state = 'selected'
-            }
-
-            const item = (
-              <TimelineItem
-                chunk={chunk}
-                isSelectionBottom={isSelectionBottom}
-                isSelectionTop={isSelectionTop}
-                key={chunk.id}
-                state={state}
-                onSelect={onSelect}
-                type={chunk.type}
-                timestamp={chunk.endTimestamp}
-              />
-            )
-
-            // Flip it back to normal after we've rendered the active one.
-            if (state === 'selected') {
-              state = 'enabled'
-            }
-
-            return item
-          })}
-        </ol>
-
-        {!timeline.reachedEarliestEntry && (
-          <div className={styles.loading} ref={setLoadingElement}>
-            <Spinner center />
-          </div>
-        )}
-      </div>
-    )
-  }
-)
+    }
+  }, [onLoadMore, loadingElement])
+
+  // Load whenever it's needed
+  useEffect(checkIfLoadIsNeeded, [checkIfLoadIsNeeded])
+
+  return (
+    <Root ref={rootRef as any} onScroll={checkIfLoadIsNeeded} data-ui="timeline">
+      {timeline.chunkCount === 0 && (
+        <Stack padding={4} space={3} style={{maxWidth: 200}}>
+          <Text weight="bold">No document history</Text>
+          <Text muted size={1}>
+            When changing the content of the document, the document versions will appear in this
+            menu.
+          </Text>
+        </Stack>
+      )}
+      <Menu ref={listRef} padding={2} space={0}>
+        {timeline.mapChunks((chunk) => {
+          const isSelectionTop = topSelection === chunk
+          const isSelectionBottom = bottomSelection === chunk
+
+          if (isSelectionTop) {
+            state = 'withinSelection'
+          }
+
+          if (isSelectionBottom) {
+            state = 'selected'
+          }
+
+          const item = (
+            <TimelineItem
+              chunk={chunk}
+              isSelectionBottom={isSelectionBottom}
+              isSelectionTop={isSelectionTop}
+              key={chunk.id}
+              state={state}
+              onSelect={onSelect}
+              type={chunk.type}
+              timestamp={chunk.endTimestamp}
+            />
+          )
+
+          // Flip it back to normal after we've rendered the active one.
+          if (state === 'selected') {
+            state = 'enabled'
+          }
+
+          return item
+        })}
+      </Menu>
+
+      {!timeline.reachedEarliestEntry && (
+        <Box padding={4} ref={setLoadingElement}>
```

Comment:
When using the `Spinner` from `@sanity/ui` we need to wrap it in a Flex to center it:

```tsx
<Flex align="center" justify="center" padding={4} ref={setLoadingElement}>
  <Spinner muted />
</Flex>
```

Reply:
Replaced 👍